bounce/db/image.py

```diff
@@ -0,0 +1,59 @@
+"""Defines an interface for storing and retreiving images."""
+
+import os
+from enum import Enum
+
+
+class EntityType(Enum):
+    """Represents types of entities for which images are stored."""
+    USER = 'user'
+    CLUB = 'club'
+
+
+def save(directory,
+         entity_type,
+         entity_id,
+         image_name,
+         image_content,
+         must_exist=False):
+    """Saves the given image and returns the path to the image."""
+    path = os.path.join(directory, entity_type.value, str(entity_id),
+                        image_name)
+    if must_exist and not os.path.exists(path):
+        raise FileNotFoundError(f'No such image {path}')
+    else:
+        # Create the directory(s) to store the file in before creating it
+        os.makedirs(os.path.dirname(path), exist_ok=True)
+    with open(path, 'wb') as image_file:
```

Comment:
this is synchronous right?

Reply:
You don't store image in a database. It's bad practice.